azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -73,20 +80,39 @@ func resourceResourceAuthorizationRead(d *schema.ResourceData, m interface{}) er
 	ctx := context.Background()
 	clients := m.(*client.AggregatedClient)
 
-	authorizedResource, projectID, err := expandAuthorizedResource(d)
+	authorizedResource, projectID, definitionID, err := expandAuthorizedResource(d)
 	if err != nil {
 		return err
 	}
 
-	if !*authorizedResource.Authorized {
-		// flatten structure provided by user-configuration and not read from ado
-		flattenAuthorizedResource(d, authorizedResource, projectID)
+	if definitionID == 0 {
+		if *authorizedResource.Authorized {
+			// (attempt) flatten read result from ado
+			resourceRefs, err := clients.BuildClient.GetProjectResources(ctx, build.GetProjectResourcesArgs{
+				Project: &projectID,
+				Type:    authorizedResource.Type,
+				Id:      authorizedResource.Id,
+			})
+
+			if err != nil {
+				return err
+			}
+
+			if len(*resourceRefs) == 0 {
+				log.Printf("[WARN] The authorization with ID '%s' no longer exists. Setting Id to empty \n", *(authorizedResource.Id))
```

Comment:
Can you add this message to line 120 as well?

![image](https://user-images.githubusercontent.com/14177833/87528235-8eff3600-c68d-11ea-96c0-d8f801f30428.png)